include/nanobind/stl/chrono.h

```diff
@@ -42,15 +42,10 @@ template <typename type> class duration_caster {
 
         // If invoked with datetime.delta object, unpack it
         int dd, ss, uu;
-        try {
-            if (unpack_timedelta(src.ptr(), &dd, &ss, &uu)) {
-                value = type(ch::duration_cast<duration_t>(
-                                 days(dd) + ch::seconds(ss) + ch::microseconds(uu)));
-                return true;
-            }
-        } catch (python_error& e) {
-            e.discard_as_unraisable(src.ptr());
-            return false;
+        if (unpack_timedelta(src.ptr(), &dd, &ss, &uu)) {
+            value = type(ch::duration_cast<duration_t>(
+                             days(dd) + ch::seconds(ss) + ch::microseconds(uu)));
+            return true;
         }
```

Comment:
```c++
#define throw if (true) {
#define catch if (false) { //
```

If this doesn't work out, we can replace the `try` with an `if (true)` so we have a smaller patchset (no whitespace changes).

Reply:
Implemented as a `sed` commit